addons/project/models/project_task.py

```diff
@@ -1116,14 +1080,14 @@ def update_date_end(self, stage_id):
     # Subtasks
     # ---------------------------------------------------
 
-    @api.depends('parent_id.partner_id', 'project_id', 'is_private')
+    @api.depends('parent_id.partner_id', 'project_id')
     def _compute_partner_id(self):
         """ Compute the partner_id when the tasks have no partner_id.
 
             Use the project partner_id if any, or else the parent task partner_id.
         """
         for task in self:
-            if task.partner_id and task.is_private:
+            if task.partner_id and not (task.project_id or task.parent_id):
```

Comment:
Hello @xavierbol and @auon-odoo 
I don't understand why ` not (task.project_id or task.parent_id)` and what Xavior suggest because if my sub-tasks  contains different customers so removing customer of parent should not remove customer of all the child are they might be part of project directly

[Here](https://github.com/odoo/odoo/pull/153391) I need to pass context `default_project_id` from sub-task actions in order to display some field like milestone and customer while creating a task and as their visibility direct depends on project
But adding that context means some of your jacks are breaking one of them is this :cry: